docs_src/microservices/device/V3Migration.md

```diff
@@ -1,315 +1,168 @@
-A `MessageQueue` section is added in configuration to specify the detail of it.
-
-!!! example "MessageQueue Example"
-```toml
-[MessageQueue]
-Protocol = "redis"
-Host = "localhost"
-Port = 6379
-Type = "redis"
-AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
-SecretName = "redisdb"
-PublishTopicPrefix = "edgex/events/device" # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
-  [MessageQueue.Optional]
-  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
-  # Client Identifiers
-  ClientId = "device-simple"
-  # Connection information
-  Qos = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
-  KeepAlive = "10" # Seconds (must be 2 or greater)
-  Retained = "false"
-  AutoReconnect = "true"
-  ConnectTimeout = "5" # Seconds
-  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
-```
-
-See the [Device Service MessageQueue](../../design/adr/013-Device-Service-Events-Message-Bus.md#device-services) section for details.
-
-### Code (Golang)
-
-#### Dependencies
-You first need to update the `go.mod` file to specify `go 1.16` and the V2 versions of the Device SDK and any EdgeX go-mods directly used by your service. Note the extra `/v2` for the modules.
-
-!!! example "Example go.mod for V2"
-
-    ```go
-    module <your service>
-    
-    go 1.16
-    
-    require (
-    	github.com/edgexfoundry/device-sdk-go/v2 v2.0.0
-    	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.0.0
-        ...
-    )
-    ```
+## Supported Device Services
 
-Once that is complete then the import statements for these dependencies must be updated to include the `/v2` in the path.
+### Device MQTT
 
-!!! example "Example import statements for V2"
+This section is specific to changes made only to **Device MQTT**. 
 
-    ```go
-    import (
-    	...
-        
-    	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
-    	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
-    )
-    ```
+See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details applicable to all EdgeX services and [All Device Services](#all-device-services) section above for details applicable to all EdgeX device services. 
+
+#### Metadata in MQTT Topics
 
-#### CommandValue
-`CommandValue` is redesigned to be more simple and straightforward. A single `Value` with `interface{}` type is able to accommodate reading value of supported type.
-As a result, you might notice the original API to create CommandValue is no longer working.
-In V2 we refactor all those API functions to create CommandValue of different type to a generic function:
+For EdgeX 3.0, Device MQTT now only supports the multi-level topics. Publishing the metadata and command/reading data wrapped in a JSON object is no longer supported. The published payload is now always only the reading data. 
```

Comment:
how about we put breaking change notation in front of it to be more visible?

Reply:
That would be redundant since the migration guide is all about breaking changes. Other enhancement are not included.

Reply:
ok